ark/mibi/qc_comp.py

```diff
@@ -1,14 +1,160 @@
 import copy
+import os
 import numpy as np
 import pandas as pd
 import matplotlib.pyplot as plt
 from scipy.ndimage import gaussian_filter
 import seaborn as sns
+from shutil import rmtree
+from skimage.io import imsave
 
+from ark.mibi.mibitracker_utils import MibiRequests
+
+import ark.settings as settings
 import ark.utils.io_utils as io_utils
 import ark.utils.load_utils as load_utils
 import ark.utils.misc_utils as misc_utils
 
+# needed to prevent UserWarning: low contrast image barf when saving images
+import warnings
+warnings.filterwarnings('ignore')
+
+
+def create_mibitracker_request_helper(email, password):
+    """Create a mibitracker request helper to access a user's MIBItracker info on Ionpath
+
+    Args:
+        email (str):
+            The user's MIBItracker email address
+        password (str):
+            The user's MIBItracker password
+
+    Returns:
+        ark.mibi.mibitracker_utils.MibiRequests:
+            A request helper module instance to access a user's MIBItracker info
+    """
+
+    return MibiRequests(settings.MIBITRACKER_BACKEND, email, password)
+
+
+def download_mibitracker_data(email, password, run_name, run_label, base_dir, tiff_dir,
+                              xml_name, img_sub_folder=None, fovs=None, channels=None):
+    """Download a specific run's image data off of MIBITracker
+    in an `ark` compatible directory structure
+
+    Args:
+        email (str):
+            The user's MIBItracker email address
+        password (str):
+            The user's MIBItracker password
+        run_name (str):
+            The name of the run (specified on the user's MIBItracker run page)
+        run_label (str):
+            The label of the run (specified on the user's MIBItracker run page)
+        base_dir (str):
+            Where to place the created `tiff_dir`
+        tiff_dir (str):
+            The name of the data directory in `base_dir` to write the run's image data to
+        xml_name (str):
+            The name of the XML file to save the run metadata to in `data_dir`.
+            Needs to be suffixed by `.xml`
+        img_sub_folder (str):
+            If specified, the subdirectory inside each FOV folder in `data_dir` to place
+            the image data into
+        fovs (list):
+            A list of FOVs to subset over. If `None`, uses all FOVs.
+        channels (lsit):
+            A list of channels to subset over. If `None`, uses all channels.
+    """
+
+    # verify that base_dir provided exists
+    if not os.path.exists(base_dir):
+        raise FileNotFoundError("base_dir %s does not exist" % base_dir)
+
+    # create the MIBItracker request helper
+    mr = create_mibitracker_request_helper(email, password)
```

Comment:
does this automatically retry if wrong pw is entered, or fails? Is the error message interpretable, or do we need to add our own?

Reply:
Ah good point, I'll add a check for this.